docs/lsm/index.rst

```diff
@@ -47,27 +47,61 @@
 
     - The name of the service entity
     - A list of attributes that have a name, a type (number, string, ...) and a modifier. This modifier determines whether the attribute is:
-        - r / readonly: readonly from the client perspective and allocated server side by the LSM
-        - rw / read-write: Attributes can be set at creation time, but are readonly after the creation
-        - rw+ / read-write always: Attribute can be set at creation time and modified when the service state allows it
+        - ``r`` / readonly: readonly from the client perspective and allocated server side by the LSM
+        - ``rw`` / read-write: Attributes can be set at creation time, but are readonly after the creation
+        - ``rw+`` / read-write always: Attribute can be set at creation time and modified when the service state allows it
     - The :ref:`fsm` that defines the lifecycle of the service.
 
     For each service entity the lifecycle service manager will create an REST API endpoint on the service inventory to perform create,
     read, update and delete (CRUD) service instances of service entities.
 
     Creating service entities
     =========================
+    Service entities are :ref:`entities <lang-entity>` that extend ``lsm::ServiceEntity``
+    We define attributes for the service entity the same way as for entities. We can also define a modifier for the attribute.
+    If no modifier is defined for an attribute, it will be ``rw`` by default.
+    Here is an example of an entity definition where the modifier of the ``address`` attribute is set to ``rw+``.
 
-    TODO:
-    How to define a new service entity. "Exported" entity is high level entity in a service model.
+    .. code-block:: inmanta
 
-    It's also possible to define a service identity for a service. For more information, see :ref:`service_identity`.
+        entity InterfaceIPAssignment extends lsm::ServiceEntity:
+            """
+                Interface details.
 
-    Updating service entities
-    =========================
+                :attr router_ip: The IP address of the SR linux router that should be configured.
+                :attr router_name: The name of the SR linux router that should be configured.
+                :attr interface_name: The name of the interface of the router that should be configured.
+                :attr address: The IP-address to assign to the given interface.
+            """
+
+            std::ipv_any_address router_ip
+            string router_name
+            string interface_name
+
+            std::ipv_any_interface address
+            lsm::attribute_modifier address__modifier="rw+"
+        end
+
+        implement InterfaceIPAssignment using parents
```

Comment:
When we include a model in the docs, we usually add the model to a separate file. Then we can use that file to add a test case that verifies whether the model actually compiles and we can use the same file to include the model (or part of it) in the docs. We should do the same here. Otherwise the code included in the docs might no longer compile due to future changes to the lsm module or compiler.